src/test/ui/parser/issue-68987-unmatch-issue-1.rs

```diff
@@ -0,0 +1,12 @@
+// This file has unexpected closing delimiter,
+
+fn func(o: Option<u32>) {
+    match o {
+        Some(_x) => {}   // Extra '}'
```

Comment:
Yeah, this is the case where I wanted to keep the span pointing at `{}`. I s there a way we can get that back?

Reply:
Current error report it's not the empty block at line 5, it's line 8:

```console
error: unexpected closing delimiter: `}`
  --> src/test/ui/parser/issue-68987-unmatch-issue-1.rs:10:1
   |
8  |         None => {}
   |                 -- block is empty, you might have not meant to close it
9  |     }
10 | } //~ ERROR unexpected closing delimiter
   | ^ unexpected closing delimiter
```

So as I thought it's not helpful enough, same comments from https://github.com/rust-lang/rust/issues/68987#issue-562129712.

I can keep the `empty block` tip, but we keep it at the one in line 5?

Reply:
I tried to bring it back, such as this case:

```rust
pub enum ErrorHandled {
    Reported,
    TooGeneric,
}

impl ErrorHandled {
    pub fn assert_reported(self) {
        match self {
            ErrorHandled::Reported => {}}
                                     //^~ ERROR block is empty, you might have not meant to close it
            ErrorHandled::TooGeneric => panic!(),
        }
    }
} //~ ERROR unexpected closing delimiter: `}`

```


After adding it back, the output is:
```console
rustc +dev src/test/ui/parser/issues/issue-70583-block-is-empty-2.rs
error: unexpected closing delimiter: `}`
  --> src/test/ui/parser/issues/issue-70583-block-is-empty-2.rs:14:1
   |
8  |         match self {
   |                    - this delimiter might not be properly closed...
9  |             ErrorHandled::Reported => {}}
   |                                       --- ...as it matches this but it has different indentation
   |                                       |
   |                                       block is empty, you might have not meant to close it
...
14 | } //~ ERROR unexpected closing delimiter: `}`
   | ^ unexpected closing delimiter
```

I think it's a little bit duplicated since we have already pointed out the candidates span for error.

Reply:
code updated, now issue-68987 will report out this:

```console
rustc +dev src/test/ui/parser/issue-68987-unmatch-issue-1.rs
error: unexpected closing delimiter: `}`
  --> src/test/ui/parser/issue-68987-unmatch-issue-1.rs:10:1
   |
4  |     match o {
   |             - this delimiter might not be properly closed...
5  |         Some(_x) => {}   // Extra '}'
   |                     -- block is empty, you might have not meant to close it
6  |             let _ = if true {};
7  |         }
   |         - ...as it matches this but it has different indentation
...
10 | } //~ ERROR unexpected closing delimiter
   | ^ unexpected closing delimiter
```